src/common/item-types/redirect-item-action.ts

```diff
@@ -0,0 +1,80 @@
+//////////////////////////////////////////////////////////////////////////////////////////
+//   _  _ ____ _  _ ___  ____                                                           //
+//   |_/  |__| |\ | |  \ |  |    This file belongs to Kando, the cross-platform         //
+//   | \_ |  | | \| |__/ |__|    pie menu. Read more on github.com/menu/kando           //
+//                                                                                      //
+//////////////////////////////////////////////////////////////////////////////////////////
+
+// SPDX-FileCopyrightText: yar2001T <https://github.com/yar2000T>
+// SPDX-License-Identifier: MIT
+
+import { IMenuItem } from '../index';
+import { IItemAction } from '../item-action-registry';
+import { DeepReadonly } from '../../main/settings';
+import { IItemData } from './redirect-item-type';
+import { Backend, WMInfo } from '../../main/backends/backend';
+import { kando } from '../../main';
+import { Notification } from 'electron';
+import * as path from 'path'; // Ensure 'path' module is imported
+
+export class RedirectItemAction implements IItemAction {
+  delayedExecution(item: DeepReadonly<IMenuItem>) {
+    return (item.data as IItemData).delayed;
+  }
+
+  async execute(
+    item: DeepReadonly<IMenuItem>,
+    _backend: Backend, // eslint-disable-line @typescript-eslint/no-unused-vars
+    _wmInfo: WMInfo // eslint-disable-line @typescript-eslint/no-unused-vars
+  ): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      try {
+        const menu = (item.data as IItemData).menu;
+
+        const env = { ...process.env };
+        delete env.CHROME_DESKTOP;
+
+        if (!menu) {
+          const errorMessage = 'Menu name is undefined or empty.';
+          console.error(errorMessage);
+          this.showError(errorMessage, 'Invalid menu name');
+          reject(new Error(errorMessage));
+          return;
+        }
+
+        try {
+          kando.showMenu({
+            trigger: '',
+            name: menu,
+          });
+
+          resolve();
+        } catch (error) {
+          const errorMessage = 'Error showing menu';
+          console.error(errorMessage + ':', error);
+          this.showError(errorMessage, error.toString());
+          reject(error);
+        }
+      } catch (error) {
+        const errorMessage = 'Unexpected error in execute';
+        console.error(errorMessage + ':', error);
+        this.showError(errorMessage, error.toString());
+        reject(error);
+      }
+    });
+  }
+
+  private showError(message: string, error: string) {
+    console.error(message + ': ' + error);
+
+    if (Notification.isSupported()) {
+      const notification = new Notification({
+        title: message + '.',
+        body: error,
+        icon: path.join(__dirname, require('../../../assets/icons/icon.png')),
+      });
+
+      notification.show();
+    }
+  }
+}
```

Comment:
I think this can be done much shorter. You should not perform the error checks here, as any thrown errors will be catched anyways by the caller of the execute method.

```suggestion
import { IItemData } from './redirect-item-type';
import { kando } from '../../main';

export class RedirectItemAction implements IItemAction {
  /**
   * Redirects are always opened immediately.
   *
   * @returns False
   */
  delayedExecution() {
    return false;
  }

  /**
   * This method opens the specified menu.
   *
   * @param item The item for which the action should be executed.
   * @returns A promise which resolves when the macro has been successfully simulated.
   */
  async execute(item: DeepReadonly<IMenuItem>) {
    const menu = (item.data as IItemData).menu;

    if (menu === '') {
      throw new Error('Menu name should not be empty!');
    }

    kando.showMenu({
      trigger: '',
      name: menu,
    });
  }
}
```